frontend/src/MainChat.tsx

```diff
@@ -1,67 +1,61 @@
-import React, {useCallback, useEffect, useRef, useState} from 'react';
-import {nanoid} from "nanoid";
+import React, {useCallback, useEffect, useState} from 'react';
 import styled from "styled-components";
+import useWebSocket, {ReadyState} from 'react-use-websocket';
 
 export default function MainChat() {
 
-    const connection = useRef<WebSocket>();
-    const shouldKeepWSAlive = useRef<boolean>(false);
-    const [messagesList, setMessagesList] = useState<any[]>([]);
+    const [messageHistory, setMessageHistory] = useState<any>([]);
+    const [message, setMessage] = useState('');
+
+    const {sendMessage, lastMessage, readyState} = useWebSocket("ws://localhost:8080/api/mainchat");
```

Comment:
Das Vorgehen mit der `lastMessage` fühlt sich ein bisschen wie ein "Hack" an. Wenn ich es richtig verstehe speicherst Du die letzte Nachricht in einem State, der dann aber nur einmalig ausgelesen wird (um in die messsageList aufgenommen zu werden).

Schöner fände ich, wenn Du einen Callback verwendest, der bei jeder neuen Nachricht die neuen Daten direkt in die Liste übernimmt. Tipp: Du kannst `useWebsocket("url", { onMessage: ...})` nutzen.

Reply:
jop fand ich cool :D
https://www.npmjs.com/package/react-use-websocket hier war das als beispiel schon mit bei :)

aber schau mal hier bitte: https://github.com/chris-yooo/capstone-chrat-reloaded/pull/8/files